src/Dispatcher.js

```diff
@@ -11,17 +14,49 @@ export default class ReduxDispatcher {
     redux: PropTypes.object.isRequired
   };
 
-  constructor(props) {
-    this.dispatcher = createDispatcher();
-    this.dispatcher.receiveStores(props.stores);
+  getChildContext() {
+    return { redux: this };
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.dispatcher.receiveStores(nextProps.stores);
+  constructor(props, context) {
+    super(props, context);
+
+    this.subscriptions = [];
+    this.emitChange = this.emitChange.bind(this);
+    this.dispatch = this.dispatch.bind(this);
+
+    const initialAtom = dispatch(props.store, undefined, {});
+    this.setAtom(initialAtom);
   }
 
-  getChildContext() {
-    return { redux: this.dispatcher };
+  dispatch(action) {
+    const nextAtom = dispatch(this.props.store, this.atom, action);
+    this.setAtom(nextAtom);
+  }
+
+  setAtom(atom) {
+    this.atom = atom;
+    if (this.state) {
+      this.setState({ atom });
```

Comment:
It’s not exactly FP-stigma.. I also love this ideas. But to bring them to JS community means creating APIs that explain these concepts instead of using strict terminology in README documents.

> On 06 Jun 2015, at 14:54 , Rafał Cieślak <notifications@github.com> wrote:
> 
> In src/Dispatcher.js <https://github.com/gaearon/redux/pull/46#discussion_r31867392>:
> 
> >    }
> >  
> > -  getChildContext() {
> > -    return { redux: this.dispatcher };
> > +  dispatch(action) {
> > +    const nextAtom = dispatch(this.props.store, this.atom, action);
> > +    this.setAtom(nextAtom);
> > +  }
> > +
> > +  setAtom(atom) {
> > +    this.atom = atom;
> > +    if (this.state) {
> > +      this.setState({ atom });
> I think these concepts are neither fancy nor complicated, but I understand your reasoning. I was once scared by those ideas, too; if avoiding the functional lingo means the library is more approachable to more people, then it's okay I guess. 
> 
> Too bad we have this FP-stigma, though. JS community could learn so much from FP (as it already did).
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/gaearon/redux/pull/46/files#r31867392>.
>